### ISSUE TITLE
Why do you use cross entropy ? 

### ISSUE THREAD

**Issue by shamanez** [RElbers/info-nce-pytorch#9]
Is it same as the following INFo-NCE loss?

![image](https://user-images.githubusercontent.com/16892570/223411372-7555e2ca-93a2-4cf2-a4bc-a55dd3838b3f.png)

**Comment by RElbers** [comment]
Yes, its the same (see https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html). The cross entropy function is usually used for classification problems, where you want your model to predict the correct class out of K classes for each sample. 
The InfoNCE loss can also be interpreted as a kind of classification, where you have K-1 negative classes and 1 positive class.

**Comment by shamanez** [comment]
Ok got it.
 
Just one more thing, I guess this loss is the same as the **NT-Xet loss** or [MultipleNegative Ranking loss](https://www.sbert.net/docs/package_reference/losses.html#multiplenegativesrankingloss) in Sentece Transformers right?

**Comment by RElbers** [comment]
I'm not familiar with MultipleNegativesRankingLoss, but it looks the same. I often see the NT-Xent loss and Info-NCE being used interchangeably, but I believe technically the difference is that NT-Xent excludes the positive sample from the denominator whereas Info-NCE includes it. In the papers there is some variation in how they get the embeddings and which (combination of) samples they use for the loss, but functionally it essentially comes down to doing softmax on embeddings.

**Comment by shamanez** [comment]
Thanks a lot. In one tutorial I’ve seen the difference in NT-dent is we use
softmax scores and a temperature.

https://youtu.be/iqzJybIk4Go?t=1408
